test/functional/p2p_node_network_limited.py

```diff
@@ -54,6 +58,63 @@ def setup_network(self):
         self.add_nodes(self.num_nodes, self.extra_args)
         self.start_nodes()
 
+    def test_avoid_requesting_historical_blocks(self):
+        self.log.info("Test full node does not request blocks beyond the limited peer threshold")
+        pruned_node = self.nodes[0]
+        miner = self.nodes[1]
+        full_node = self.nodes[2]
+
+        # Connect and generate block to ensure IBD=false
+        self.connect_nodes(1, 0)
+        self.connect_nodes(1, 2)
+        self.generate(miner, 1)
```

Comment:
> I don't understand this part of the test. Why do we connect nodes 1 and 2, just to disconnect again without doing anything in between? Why is the generate call necessary?

To ensure all nodes are out of IBD. The idea was to isolate this test case from the previous cases executed along the file. It was done in this way to be able to run this test case at any point of the file. This was vasild request because he was running it first, commenting the rest of the cases to speed the process up.

The connection requirement is because the test overrides `setup_network`, which skips the framework's connections creation process (peers start the test with no connections).

> shouldn't we then also at least add a sync_blocks to ensure it propagates to all nodes before disconnecting?

The `self.generate(miner, 1)` function calls `sync_blocks()` internally.

Reply:
ok, makes sense!